Standardise component structure (#150)
sa-tre/satre-specification: docs/source/pillars/computing_technology.md

    In practice a web browser is the most common way to achieve this.
  - Recommended
* - Your TRE workspace should provide an environment familiar to your users.
  - This may take the form of a virtual Windows or Linux desktops, non-desktop interfaces such as JupyterLab and other web applications, or a terminal.
    Bespoke TRE-specific software should be avoided when widely used alternatives already exist.
  - Recommended
* - Your TRE should take accessibility for users with disabilities into account.
  - The restricted nature of TREs means many assistive tools such as screenreaders in a virtual desktop may not be allowed, but other options such as colour schemes, font sizes, and resizing user interface elements, should be supported.
  - Recommended
* - You should disable the ability to copy data out of your TRE via the system clipboard.
  - A TRE user must not be able to copy sensitive data out of a workspace using the system clipboard.
    A TRE may allow user to paste text into a workspace.
  - Recommended
* - A TRE could restrict data access from researchers entirely and provide an interface for submitting code.

>   - For example, you might use a system where users submit jobs which run over the data and return results without allowing direct data acces.

Reviewer: ```suggestion
  - For example, you might use a system where users submit jobs that run over the data and return results without allowing direct data access.
```

    The set of tools should be reviewed regularly to ensure they are up to date.
  - Mandatory
* - Your TRE must provide clear guidance on how to use software tools and work with data in the TRE.
  - TREs that provide a virtual desktop environment for researchers to work in should provide documentation detailing the available tools.
    TREs where the analysis code is developed on the access machine (as opppose to within the TRE) should provide documentation detailing the mechanism by which code is submitted to the TRE.
  - Mandatory
* - Your TRE should provide tools to encourage best-practice in reproducibly analysing data.
  - Reproducibility of analyses improves auditability and accountability of how data has been used, as well as being best-practice in research.
    This may include version control software, and tools for developing and running data analysis pipelines.
  - Recommended
* - Your TRE should, where possible, automatically apply security related updates for user software.

>   - Reducing the risk of expoitable vulnerabilities in installed software will increase the security of your TRE.

Reviewer: ```suggestion
  - Reducing the risk of exploitable vulnerabilities in installed software will increase the security of your TRE.
```

  - Recommended
* - Your TRE could provide shared services that are accessible to users in the same project.
  - This may include shared file storage, databases, collaborative writing, and other web applications.
    This must only be shared amongst users within the same project.
  - Optional
* - Your TRE could provide limited access to some public software repositories or container registries.
  - For example, a TRE may allow installation of packages from Python or R repositories, or provide an internal mirror with approved packages.
    Similarly a subset of public containers could be made available, or individual container images via an internal container registry.
  - Optional
* - Your TRE could include licenced commercial software if required by researchers, but additional risks must be recorded and mitigated where neccesary.

>   - For example, if an application must connect to an external licensing server TREs must be confident that only licensing information is sent to this server, and that any network proxies to allow this are secured.

Reviewer: ```suggestion
  - For example, if an application must connect to an external licensing server, you  must be confident that only licensing information is sent to this server, and that any network proxies to allow this are secure.
```

* - Statement
  - Guidance
  - Importance
* - Your TRE must record usage data.
  - This may include the number of users, number of projects, the amount of data stored, number of datasets, the number of workspaces, etc.
  - Mandatory

> * - Your TRE should record which datasets are accessed, when and by who.

Reviewer: ```suggestion
* - Your TRE should record which datasets are accessed, when and by whom.
```

* - Your TRE must record usage data.
  - This may include the number of users, number of projects, the amount of data stored, number of datasets, the number of workspaces, etc.
  - Mandatory

> * - Your TRE should record which datasets are accessed, when and by whom.

Reviewer 2: ```suggestion
* - Your TRE should record which datasets are accessed, when and by who.
```

@jemrobinson @craddm How strongly do you feel about that?

It isn't necessary to use whom here and it feels a little anachronistic.

Author: I think @craddm suggested it and it *is* correct, so I'd probably leave it. If you prefer to change it back I'm not going to complain though.

Reviewer 2: Not incorrect, but I do think for modern English it is unusual and comes across as stuffy and possibly a little pretentious.
